ui/src/components/CubePreview.vue

```diff
@@ -84,34 +84,41 @@
       <tfoot>
         <tr>
           <td :colspan="tableWidth" class="has-background-light">
-            <div class="is-flex gap-4">
+            <div class="is-flex is-justify-content-space-between gap-4">
               <div class="is-flex is-align-items-center gap-1">
-                <span class="pr-1">Page {{ page }}</span>
                 <b-tooltip label="Previous page">
                   <b-button
                     icon-left="chevron-left"
                     @click="page = page - 1"
-                    :disabled="!observations.data || page === 1"
+                    :disabled="!hasPreviousPage"
                   />
                 </b-tooltip>
                 <b-tooltip label="Next page">
                   <b-button
                     icon-left="chevron-right"
                     @click="page = page + 1"
-                    :disabled="!observations.data || observations.data.length === 0"
+                    :disabled="!hasNextPage"
                   />
                 </b-tooltip>
+                <span class="ml-4">Page</span>
+                <b-input v-model.number="page" type="number" class="is-inline-block w-20" />
```

Comment:
The binding causes immediate reload on typing. It should be deferred or happen on `blur`. Right now if I type, say, 6800, there will be requests or pages `6`, `68`, `680` and `6800` :)

Reply:
Good catch